### ISSUE TITLE
UserProfileManager:create_user signature does not match django's default manager

### ISSUE THREAD

**Issue by He3lixxx** [fsr-de/EvaP#1368]
Django's default user manager ([docs](https://docs.djangoproject.com/en/2.2/ref/contrib/auth/#manager-methods)) has this signature for `create_user`:
```python
def create_user(self, username, email=None, password=None, **extra_fields):
```

Currently, our `UserProfileManager` has the argument order swapped, which can lead to nasty bugs if it is called assuming the other signature:
```python
def create_user(self, username, password=None, email=None, first_name=None, last_name=None):
```

A quick grep for `create_user` indicates that this method is not actually used in our code (except in `create_superuser`, which itself seems unused).

Do we need this method? Is it called by some django code (admin interface, manage.py, ...)?

If we don't need it, I guess we should remove it. If we need it, we should consider reordering the arguments, maybe even making them keyword arguments so that you can not give `password` as a positional argument.

**Comment by karyon** [comment]
https://docs.djangoproject.com/en/2.2/topics/auth/customizing/#writing-a-manager-for-a-custom-user-model says you should provide those methods, though it does not say why. at least for create_superuser there is the management commend createsuperuser which is using it: https://github.com/django/django/blob/03dbdfd9bbbbd0b0172aad648c6bbe3f39541137/django/contrib/auth/management/commands/createsuperuser.py#L189

**Comment by He3lixxx** [comment]
The django documentation also says something about how the prototype should look like:
https://docs.djangoproject.com/en/2.2/topics/auth/customizing/#django.contrib.auth.models.CustomUserManager.create_user

> The prototype of create_user() should accept the username field, plus all required fields as arguments. For example, if your user model uses email as the username field, and has date_of_birth as a required field, then create_user should be defined as:
>
>     def create_user(self, email, date_of_birth, password=None):
>         # create user here
>        ...

So just swapping `username` and `password` in the signature should be fine. The bug we encountered is in mozilla-django-oidc. Our code matches the specification. I'll open a bug issue for them.

**Comment by He3lixxx** [comment]
Solved in #1408
